llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -4888,14 +4888,9 @@ static bool getUniformBase(const Value *Ptr, SDValue &Base, SDValue &Index,
 
   assert(Ptr->getType()->isVectorTy() && "Unexpected pointer type");
 
-  // Handle splat constant pointer.
-  if (auto *C = dyn_cast<Constant>(Ptr)) {
-    C = C->getSplatValue();
-    if (!C)
-      return false;
-
-    Base = SDB->getValue(C);
-
+  // Handle splat (possibly constant) pointer.
+  if (Value *ScalarV = getSplatValue(Ptr)) {
```

Comment:
Does this work if the splat instruction isn't in the same basic block? You have to be careful looking through instructions in SelectinDAGBuilder because the Value* won't be exported from the producing basic block.

Reply:
Oh yuck, I should have checked the history here more closely.  No, it doesn't.  

I'm going to close this and revisit from scratch.  I can probably find an analogous approach, but TBD.